src/server.rs

```diff
@@ -686,83 +532,195 @@ pub fn generate_server_in_order_trait(
+                                    idol_runtime::RequestError::Runtime(e) => {
+                                        let mut reply_buf = [0u8; <#ty as hubpack::SerializedSize>::MAX_SIZE];
+                                        let n_reply = hubpack::serialize(&mut reply_buf, &e).map_err(|_| ()).unwrap_lite();
+                                        userlib::sys_reply(rm.sender, 1, &reply_buf[..n_reply]);
+                                    }
+                                }
+                                Ok(())
+                            },
+                            encoding => panic!("Complex error types not supported for {encoding:?} encoding"),
+                        },
+                        syntax::Error::CLike(_) => quote! {
+                            Err(val.map_runtime(u16::from))
+                        },
+                        syntax::Error::ServerDeath => quote! {
+                            Err(val.map_runtime(|e| match e {}))
+                        }
+                    };
+                    quote! {
+                        match r {
+                            Ok(val) => {
+                                #encode
+                                Ok(())
+                            }
+                            Err(val) => {
+                                #err_case
                             }
-                            _ => panic!("Complex error types not supported for {:?} encoding", op.encoding),
                         }
-                        writeln!(out, "                        Ok(())")?;
                     }
-                    syntax::Error::CLike(_) => {
-                        writeln!(
-                            out,
-                            "                        Err(val.map_runtime(u16::from))"
-                        )?;
+                }
+            }
+        };
+        quote! {
+            #enum_name::#opname => {
+                #check_allowed
+                #read
+                let r = self.1.#opname(
+                    rm,
+                    #( #args )*
+                    #( #leases )*
+                );
+                #reply
+            }
+        }
+    });
+    Ok(quote! {
+        #trait_def
+
+        #[automatically_derived]
+        impl <S: #trt> idol_runtime::Server<#enum_name> for (core::marker::PhantomData<#enum_name>, &'_ mut S) {
+            fn recv_source(&self) -> Option<userlib::TaskId> {
+                <S as #trt>::recv_source(self.1)
+            }
+
+            fn closed_recv_fail(&mut self) {
+                <S as #trt>::closed_recv_fail(self.1)
+            }
+            fn handle(
+                &mut self,
+                op: #enum_name,
+                incoming: &[u8],
+                rm: &userlib::RecvMessage,
+            ) -> Result<(), idol_runtime::RequestError<u16>> {
+                #[allow(unused_imports)]
+                use core::convert::TryInto;
+                #[allow(unused_imports)]
```

Comment:
Nit: this unused_imports was not in the original and should not be required.

Reply:
...alternatively... it kind of looks like you've newly qualified all references to ClientError? In which case this use might be removed.

Reply:
Ah, whoops, I didn't notice that; I can either remove the `use` or change them back --- have you a preference?